app/services/process_ping_things_service.rb

```diff
@@ -16,9 +16,18 @@ def call
         network: raw.network,
         created_at: raw.created_at
       )
+      ping_thing = PingThing.create(params)
+
+      # create sidekiq job to update stats if it's not a latest data
+      if raw.should_recalculate_stats_after_processing?
+        RecalculatePingThingStatsWorker.perform_async({
+          reported_at: ping_thing.reported_at,
+          network: ping_thing.network
+        })
+      end
```

Comment:
All good, except this doesn't tell us if `create` succedeed or not (ping_thing object will always be created, no matter if db record was created or not). And we should only create the worker if PingThing db record was created (eg. all required params like signature etc. are present and valid).
```suggestion
      ping_thing = PingThing.new(params)

      # create sidekiq job to update stats if it's not a latest data
      if ping_thing.save && raw.should_recalculate_stats_after_processing?
        RecalculatePingThingStatsWorker.perform_async({
          reported_at: ping_thing.reported_at,
          network: ping_thing.network
        })
      end
```